scan-command-test-utils/src/main/java/io/ballerina/scan/test/ScanAssertions.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.scan.test;
+
+import io.ballerina.scan.Issue;
+import io.ballerina.scan.Rule;
+import io.ballerina.scan.RuleKind;
+import io.ballerina.scan.Source;
+import org.testng.Assert;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Test utility class to assert the issues found during a scan.
+ *
+ * @since 0.10.0
+ */
+public class ScanAssertions {
+
+    private ScanAssertions() {
+    }
+
+    /**
+     * Assert that the list of issues contains an issue with the given rule ID, file name, start line, and end line.
+     *
+     * @param issues    list of issues from which the issue should be found
+     * @param ruleId    rule ID to check for
+     * @param fileName  file name to check for
+     * @param startLine start line to check for
+     * @param endLine   end line to check for
+     * @param source    source to check for
+     */
+    public static void assertHasIssue(List<Issue> issues,
```

Comment:
At least for compiler diagnostics, we usually assert in order. That way we can make sure any and all diagnostics are asserted. 

Can't we just do

```suggestion
    public static void assertIssue(Issue issue, 
```

Reply:
Similar assertion I did recently for the np module - https://github.com/ballerina-platform/module-ballerina-np/blob/main/compiler-plugin-tests/src/test/java/io/ballerina/lib/np/compilerplugintests/CompilerPluginTest.java#L80

Reply:
+1. Will change this accordingly. 

The only issue is that we are asserting in an order, although we are not guranteeing any particular order in the issue list.

Reply:
Updated with 2a93dd7c0241a36af95939e6683291de1795afe4